flink-connectors/flink-connector-jdbc/src/test/java/org/apache/flink/connector/jdbc/internal/JdbcFullTest.java

```diff
@@ -104,6 +107,52 @@ public void testEnrichedClassCastException() {
 		}
 	}
 
+	@Test
+	public void testJdbcBatchingOutputFormatCloseDuringRuntime() throws Exception{
+		JdbcOptions options = JdbcOptions.builder()
+			.setDBUrl(getDbMetadata().getUrl())
+			.setTableName(OUTPUT_TABLE)
+			.build();
+
+		RuntimeContext context = Mockito.mock(RuntimeContext.class);
+		ExecutionConfig executionConfig = Mockito.mock(ExecutionConfig.class);
+		JdbcExecutionOptions jdbcExecutionOptions = Mockito.mock(JdbcExecutionOptions.class);
+		JdbcBatchStatementExecutor executor = Mockito.mock(JdbcBatchStatementExecutor.class);
+
+		doReturn(executionConfig).when(context).getExecutionConfig();
+		// use scheduledThreadPool
+		doReturn(500L).when(jdbcExecutionOptions).getBatchIntervalMs();
+		doReturn(2).when(jdbcExecutionOptions).getBatchSize();
+		doReturn(3).when(jdbcExecutionOptions).getMaxRetries();
```

Comment:
Mockito is a bad practice which should be avoided as much as possible. You can 

```java
JdbcExecutionOptions jdbcExecutionOptions = JdbcExecutionOptions.builder()
			.withBatchIntervalMs(1000_000L)
			.withBatchSize(1)
			.withMaxRetries(1)
			.build();
ExecutionConfig executionConfig = new ExecutionConfig();
```

to create them.

Reply:
I suggest to suse 1000s for the interval and 1 for the batch to quickly throw the exception and avoid a async exception.

Reply:
If we want to use scheduler in batchingOutputFormat, we have to set the batchsize larger than 1.  So I think we setting batsize 2 is better.